src/git/reference.mli

```diff
@@ -21,6 +21,31 @@ type t = private string
 (** A Git Reference object. Which contains a hash to point to an other
     object. *)
 
+module Infix : sig
+  type partial
+  (** Type of the left part of a reference. *)
+  type branch = string
+  (** Tyoe of the right part of a reference. *)
+
+  val ( // ): partial -> string -> partial
+  (** Infix operator to compose [string] value with a {!partial} reference. *)
+
+  val ( / ): partial -> branch -> t
+  (** Infix operator to compose {!partial} with {!branch} and return a full-defined reference {!t}. *)
+
+  val refs: partial
```

Comment:
this is not really `Infix`, right? Maybe just move all the code to the top-level so that a user coud do:

```
Git.Reference.(heads / master)
```

Reply:
or name it `Path` instead of `Infix` (to not conflict with other top-level definitions)